### PR TITLE
ENG-1074: Fix XXE and XSS vulnerabilities

### PR DESCRIPTION
**Pull request by w-caffiero-entando**

### CODE REVIEW

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **w-caffiero-entando**

If they are not already assigned, you can assign the PR to them by writing `/assign @w-caffiero-entando` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fentando-plugin-jacms).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/entando/entando-plugin-jacms/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by entando-jx** [comment]
@w-caffiero-entando: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | 755a20f72868ce6277234a2b131592a99cbc6709 | [link](https://raw.githubusercontent.com/entando/devops-results/logs/jenkins-x/logs/entando/entando-plugin-jacms/PR-28/6.log) | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by w-caffiero-entando** [comment]
CodeSmells are on old code that was not touched by these changes.

File: src/main/java/com/agiletec/plugins/jacms/aps/system/services/contentmodel/ContentModel.java
Comment (eugeniosant) [reply src/main/java/com/agiletec/plugins/jacms/aps/system/services/contentmodel/ContentModel.java:148]: For me ok

File: src/main/java/com/agiletec/plugins/jacms/aps/system/services/contentmodel/ContentModel.java
Comment (eugeniosant) [reply src/main/java/com/agiletec/plugins/jacms/aps/system/services/contentmodel/ContentModel.java:163]: Ok also for that

File: src/test/java/com/agiletec/plugins/jacms/aps/system/services/contentmodel/ContentModelTest.java
Comment (eugeniosant) [reply src/test/java/com/agiletec/plugins/jacms/aps/system/services/contentmodel/ContentModelTest.java:5]: @w-caffiero-entando We must verify the case where the modelId parameter is null because in this case the system (the "Content Viewer" widget in particular) uses the default model

File: src/main/java/com/agiletec/plugins/jacms/aps/tags/ContentInfoTag.java
Comment (eugeniosant) [reply src/main/java/com/agiletec/plugins/jacms/aps/tags/ContentInfoTag.java:114]: @w-caffiero-entando see the [comment before](https://github.com/entando/entando-plugin-jacms/pull/28#discussion_r483704299)

File: src/test/java/com/agiletec/plugins/jacms/aps/system/services/contentmodel/ContentModelTest.java
Comment (w-caffiero-entando) [reply src/test/java/com/agiletec/plugins/jacms/aps/system/services/contentmodel/ContentModelTest.java:5]: ok, so <null> should be an acceptable value for isValidModelIdParam

**Comment by w-caffiero-entando** [comment]
Code Coverage stat is wrong.
Of the 12 uncovered condition in the ContentModel 0% are actually uncovered.
Tested locally with intellij coverage tool.

**Comment by entando-jx** [comment]
@w-caffiero-entando: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | f6ba976d75e454f869cc9a0c234427d796ae8dd5 | [link](https://raw.githubusercontent.com/entando/devops-results/logs/jenkins-x/logs/entando/entando-plugin-jacms/PR-28/19.log) | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by entando-jx** [comment]
@w-caffiero-entando: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | c678ab58fa4b28a9e29da8f359946f2b27cb7d7c | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by entando-jx** [comment]
@w-caffiero-entando: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | fdbb0e6f1ea7a3488e614cb1a6e9a2160b52fa42 | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by entando-jx** [comment]
@w-caffiero-entando: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | 0ecde77339751a94e61acc6affb4a44e771f8374 | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Review by eugeniosant — APPROVED** [review]

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/entando/entando-plugin-jacms/pull/28#pullrequestreview-508178426" title="Approved">eugeniosant</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fentando-plugin-jacms).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/entando/entando-plugin-jacms/blob/master/OWNERS)~~ [eugeniosant]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
